loadtest/control/simulcontroller/actions.go

```diff
@@ -1767,6 +1813,109 @@ func (c *SimulController) updateThreadRead(u user.User) control.UserActionRespon
+		return control.UserActionResponse{Info: "no team set"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	err = u.GetDrafts(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("viewed drafts for user id %v in team id %v", userId, team.Id)}
+}
+
+func (c *SimulController) upsertDraft(u user.User) control.UserActionResponse {
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	var rootId = ""
+
+	// 33% of the time draft will be a thread reply
+	if rand.Float64() < 0.33 {
+		post, err := u.Store().RandomPostForChannel(channel.Id)
+		if errors.Is(err, memstore.ErrPostNotFound) {
+			return control.UserActionResponse{Info: fmt.Sprintf("no posts found in channel %v", channel.Id)}
+		} else if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		if post.RootId != "" {
+			rootId = post.RootId
+		} else {
+			rootId = post.Id
+		}
+	}
+
+	if err := sendTypingEventIfEnabled(u, channel.Id); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	message, err := createMessage(u, channel, false)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	draft := &model.Draft{
+		Message:   message,
+		ChannelId: channel.Id,
+		RootId:    rootId,
+		CreateAt:  time.Now().Unix() * 1000,
+	}
+
+	// 2% of the times post will have files attached.
+	if rand.Float64() < 0.02 {
+		if err := c.attachFilesToDraft(u, draft); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	err = u.UpsertDraft(channel.TeamId, draft)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("draft created in channel id %v", channel.Id)}
+}
+
+func (c *SimulController) deleteDraft(u user.User) control.UserActionResponse {
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	team, err := u.Store().CurrentTeam()
+	if errors.Is(err, memstore.ErrTeamStoreEmpty) {
+		return control.UserActionResponse{Info: "no team set"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	draftId, err := u.Store().RandomDraftForTeam(team.Id)
+	if errors.Is(err, memstore.ErrDraftNotFound) {
+		return control.UserActionResponse{Info: fmt.Sprintf("no drafts found in team %v", team.Id)}
+	}
+
+	err = u.DeleteDraft(channel.Id, draftId)
```

Comment:
Here, `channel.Id` and draftId` may be the same ID if the draft is a root post, right? I understand that is correct, but I wanted to double-check, this rootid-channelid for drafts was a bit confusing for me at first.

Reply:
> Here, channel.Id and draftId` may be the same ID if the draft is a root post, right?

It's the opposite. If a draft doesn't have a root post, then draftID is same as channelID.